src/commands/createFunction/createFunction.ts

```diff
@@ -203,17 +206,18 @@ async function promptForTemplate(functionAppPath: string, language: ProjectLangu
 
     let template: IFunctionTemplate | undefined;
     while (!template) {
-        const templateProvider: TemplateProvider = await ext.templateProviderTask;
-        const templates: IFunctionTemplate[] = await templateProvider.getTemplates(language, runtime, functionAppPath, templateFilter, telemetryProperties);
-        let picks: IAzureQuickPickItem<IFunctionTemplate | string>[] = templates.map((t: IFunctionTemplate) => { return { data: t, label: t.name, description: '' }; });
-        picks = picks.concat([
-            { label: localize('selectRuntime', '$(gear) Change project runtime'), description: localize('currentRuntime', 'Current: {0}', runtime), data: runtimePickId, suppressPersistence: true },
-            { label: localize('selectLanguage', '$(gear) Change project language'), description: localize('currentLanguage', 'Current: {0}', language), data: languagePickId, suppressPersistence: true },
-            { label: localize('selectFilter', '$(gear) Change template filter'), description: localize('currentFilter', 'Current: {0}', templateFilter), data: filterPickId, suppressPersistence: true }
-        ]);
-
-        const placeHolder: string = templates.length > 0 ? localize('azFunc.selectFuncTemplate', 'Select a function template') : localize('azFunc.noTemplatesFound', 'No templates found. Change your settings to view more templates');
-        const result: IFunctionTemplate | string = (await ext.ui.showQuickPick(picks, { placeHolder })).data;
+        const picksTask: Promise<IAzureQuickPickItem<IFunctionTemplate | string>[]> = ext.templateProviderTask.then(async templateProvider => {
```

Comment:
Why `.then` this when there's a `while` loop wrapped around this rather than just using an `await`?

Reply:
If I did await here, the quick pick wouldn't have the little progress bar. We need to pass in the full promise to `showQuickPick` without awaiting